storage/watchable_store.go

```diff
@@ -241,6 +242,71 @@ func (s *watchableStore) syncWatchingsLoop() {
 	}
 }
 
+// RangeAllUnsynced ranges on them all
+// finding minimum and maximum current revision index.
+func (s *watchableStore) RangeAllUnsynced() (revbs [][]byte, kvs []storagepb.KeyValue, nextRev int64, err error) {
+	totalLimit := 0
+	minRev, maxRev := int64(math.MaxInt64), int64(math.MinInt64)
+	startKey, endKey := make([]byte, 0), make([]byte, 0)
+	for w := range s.unsynced {
+		totalLimit += cap(w.ch) - len(w.ch)
+		if minRev >= w.cur {
+			minRev = w.cur
+			startKey = w.key
+		}
+		if maxRev <= w.cur {
+			maxRev = w.cur
+			endKey = w.key
+		}
+	}
+
+	fmt.Println("startKey:", string(startKey))
+	fmt.Println("endKey:", string(endKey))
+	fmt.Println("minRev:", minRev)
+	fmt.Println("maxRev:", maxRev)
+	fmt.Println("totalLimit:", totalLimit)
+
+	// startRev must be:
+	// s.store.compactMainRev < startRev <= s.store.currentRev.main
+	startRev := minRev
+
+	if startRev > 0 && startRev <= s.store.compactMainRev {
+		return nil, nil, 0, ErrCompacted
+	}
+	if startRev > s.store.currentRev.main {
+		return nil, nil, 0, ErrFutureRev
+	}
+
+	revs := s.store.kvindex.RangeSince(startKey, endKey, startRev)
```

Comment:
we basically should not use this interface. we want to scan over all events from startRev to Now. And we do not want to filter out any keys.